pkg/util/config/feature_config.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// known keys
+	enableAllAlphaFeatures = "enableAllAlphaFeatures"
+)
+
+var featureDefaults = map[string]bool{
+	enableAllAlphaFeatures: false,
+}
+
+type FeatureConfig interface {
```

Comment:
can we s/Config/Gate/ ?  These are typically known as "flag gates" and "config" is just soooo generic